python/tank/platform/engine.py

```diff
@@ -1736,12 +1819,17 @@ def start_engine(engine_name, tk, context):
 
     # Instantiate the engine
     class_obj = loader.load_plugin(plugin_file, Engine)
-    obj = class_obj(tk, context, engine_name, env)
+    engine = class_obj(tk, context, engine_name, env)
 
     # register this engine as the current engine
-    set_current_engine(obj)
+    set_current_engine(engine)
 
-    return obj
+    # if the engine supports logging metrics, initialize the workers
+    metrics_dispatch_queue = MetricsDispatchQueue()
```

Comment:
Maybe this is a dumb question, but any reason we are not just instantiating a metrics dispatch queue in `pre_app_init()`? That way we don't need the global object? (Sorry if I am missing something obvious!)